sway/desktop/render.c

```diff
@@ -1299,6 +1299,41 @@ void output_render(struct sway_output *output, struct timespec *when,
 		fx_renderer_clear((float[]){1, 1, 0, 1});
 	}
 
+	if (server.session_lock.locked) {
+		float clear_color[] = {0.0f, 0.0f, 0.0f, 1.0f};
+		if (server.session_lock.lock == NULL) {
+			// abandoned lock -> red BG
+			clear_color[0] = 1.f;
+		}
+		int nrects;
+		pixman_box32_t *rects = pixman_region32_rectangles(damage, &nrects);
+		for (int i = 0; i < nrects; ++i) {
+			scissor_output(wlr_output, &rects[i]);
+			wlr_renderer_clear(renderer, clear_color);
```

Comment:
Here I think it may be:
```suggestion
			fx_renderer_clear(clear_color);
```

What do you think?

Reply:
I'll check on this after the 🎉 weekend :)